### ISSUE TITLE
Support shells other than Bash

### ISSUE THREAD

**Issue by iJebus** [ChaosinaCan/vscode-remote-x11#1]
So this is probably a bad on my part but something seems to be hanging; Remote X11 output shows the below:

```sh
Setting up display for remote "ssh-remote".
Connecting to SSH liam@host 22
```

Side note, I'm guessing `host` should be a variable here:

https://github.com/ChaosinaCan/vscode-remote-x11/blob/d40a64b099feb77a58d01ac605db825cd714b84c/extension/src/extension.ts#L81

Anyway, this never times out. Meanwhile Remote X11 (SSH) shows:

```sh
Connecting to liam@192.168.56.101 port 22
Connection ready. Setting up display...
...timed out.
```

If I SSH in with putty with X11, grab the DISPLAY value from that and drop it into a vscode terminal, then it's happy. Any ideas what the beef would be?

**Comment by ChaosinaCan** [comment]
You're right. That should be a `${host}`.

Do you have key-based authentication set up? It might be stuck waiting for a password (which you can't give it).

**Comment by iJebus** [comment]
Oops, thought I responded to this 😬. Yeah I'm using key based auth and have the correct private key defined in settings.

I played around with the path to private key to make sure it was actually right. When it's incorrectly defined (to incorrect key or to a path that doesn't exist) it sits on connecting, while when defined to the correct key it does the timeout.

**Comment by ChaosinaCan** [comment]
I'm not sure what's going on, but I have an idea on how to find out. Either grab the latest source or use this build (rename to .vsix) and check the Remote X11 (SSH) logs. It should now log all the output from the SSH connection.

[remote-x11-ssh-1.0.0.vsix.zip](https://github.com/ChaosinaCan/vscode-remote-x11/files/4217439/remote-x11-ssh-1.0.0.vsix.zip)

I'm using a really dumb regex to search for the echo output, so I wouldn't be surprised if your shell is configured to format things differently enough that it misses it.

**Comment by iJebus** [comment]
Yeah that's it; I'm using ZSH/[Starship](https://github.com/starship/starship). Disabling Starship didn't resolve, but changing the default shell to Bash did. 

Here's the complete ZSH output.

```sh
Connecting to liam@192.168.56.101 port 22
Connection ready. Setting up display...
Last login: Thu Feb 20 11:29:26 2020 from 192.168.56.1


echo $DISPLAY

liam@manjaro Linux 4.19.102-1-MANJARO x86_64 18.1.5 Juhraya

[1m
[0mh
o
 
$
D
I
S
P
L
A
Y
[13D[32me[32mc[32mh[32mo[39m[9C
[?2004l


localhost:10.0
[1m
[0m...timed out.
```

And successful output under bash:

```sh
Connecting to liam@192.168.56.101 port 22
Connection ready. Setting up display...
Last login: Thu Feb 20 11:45:35 2020 from 192.168.56.1


echo $DISPLAY

[01;32m[liam@manjaro[01;37m ~[01;32m]$[00m 
echo $DISPLAY
localhost:10.0

Display ready: localhost:10.0
```

Interestingly I can set the default shell to bash but override to ZSH via `"terminal.integrated.shell.linux"` and it still works, so I can keep my preferred shell/prompt :)

**Comment by ChaosinaCan** [comment]
Oh, weird... It responded with each character in separate events. I think I handle that correctly, but the big chunk of control characters in between $DISPLAY and the response is probably screwing up the regex. I'll have to find a better way to handle that.

**Comment by ChaosinaCan** [comment]
@iJebus I just published version 1.1.0 which runs the command through Bash by default or lets you change the command to work with your shell. If you're still using this extension, let me know whether that works for you.

**Comment by iJebus** [comment]
@ChaosinaCan my apologies, I've formatted and my setup has changed accordingly so nothing is set up to test this out. Thanks for doing this, I'm sure it'll help others out down the road :)
